src/backend/access/heap/heapam.c

```diff
@@ -2007,7 +2007,7 @@ heap_insert(Relation relation, HeapTuple tup, CommandId cid,
 	 * the heaptup data structure is all in local memory, not in the shared
 	 * buffer.
 	 */
-	CacheInvalidateHeapTuple(relation, heaptup, NULL);
+	CacheInvalidateHeapTuple(relation, heaptup, NULL, false);
```

Comment:
this change is very intrusive on engine side. We should think of alternatives where we have minimal changes on code outside query environment.

Reply:
agree that we should try to make this change as unintrusive as possible. Main issue is that most of the abstraction around inval messages doesn't have enough knowledge of whether it is being invoked on a temp-table related invalidation or not; so our code will likely need to touch many layers to work regardless of the approach.